pkg/generate/code/set_resource.go

```diff
@@ -302,6 +317,39 @@ func SetResource(
 	return out
 }
 
+// GetWrapperOutputShape returns the shape of the last element of a given field
+// Path. It carefully unwrapps the output shape and verifies that every element
+// of the field path exists in their correspanding parent shape and that they are
+// of type 'structure'.
+func GetWrapperOutputShape(
+	shape *awssdkmodel.Shape,
+	fieldPath string,
+) (*awssdkmodel.Shape, error) {
+	fieldPathParts := strings.Split(fieldPath, ".")
+mainLoop:
+	for _, wrapperField := range fieldPathParts {
+		for memberName, memberRef := range shape.MemberRefs {
+			if memberName == wrapperField {
+				if memberRef.Shape.Type != "structure" {
+					// All the mentionned shapes must a be of type 'structure'
+					return nil, fmt.Errorf(
+						"Expected SetInput.WrapperFieldPath to only contain fields of type 'structure'."+
+							" Found %s of type '%s'",
+						memberName, memberRef.Shape.Type,
+					)
+				}
+				shape = memberRef.Shape
+				continue mainLoop
+			}
+		}
+		return nil, fmt.Errorf(
+			"Incorrect SetInput.WrapperFieldPath. Could not find %s in Shape %s",
+			wrapperField, shape.ShapeName,
+		)
+	}
+	return shape, nil
+}
```

Comment:
I'm with @RedbackThomson on the goto label use here being a bit of a code smell...

How about using recursion instead?

```go
func GetWrapperOutputShape(
    shape *awssdkmodel.Shape,
    fieldPath string,
) (*awssdkmodel.Shape, error) {
    fieldPathParts := strings.Split(fieldPath, ".")
    for x, field := range fieldPathParts {
        for memberName, memberRef := range shape.MemberRefs {
            if memberName == wrapperField {
                if memberRef.Shape.Type != "structure" {
                    // handle error...
                }
                remainPath := strings.Join(fieldPathParts[x+1:], ".")
                return GetWrapperOutputShape(memberRef.Shape, remainPath)
        }
    }
    return shape, nil
}
```